themes/ropensci/layouts/partials/comments.html

```diff
@@ -0,0 +1,38 @@
+<!-- discourse comments -->
+<!-- 
+group 1. posts before discourse came along - no discourse embed code
+group 2. posts that have topicid in their yaml header - get discourse embed code with topocid
+group 3. posts we're trying to automate embeds for, no topicid and newer than 2019
+  (to differentiate from older posts that also don't have topicid in their yaml)
+-->
+
+{{ with .topicid }}
+<div class="col-offset-2 col-8 top-4">
+  <div id='discourse-comments'></div>
+    <script type="text/javascript">
+      DiscourseEmbed = { discourseUrl: 'https://discuss.ropensci.org/',
+        topicId: {{ . }} };
+      (function() {
+        var d = document.createElement('script'); d.type = 'text/javascript'; d.async = true;
+        d.src = DiscourseEmbed.discourseUrl + 'javascripts/embed.js';
+        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(d);
+      })();
+      </script>
+        </div>
+        {{ else }}
+      {{ if (gt .date.Year 2019) }}
+      <div class="col-offset-2 col-8 top-4">
+        <div id='discourse-comments'></div>
+        <script type="text/javascript">
+        DiscourseEmbed = { discourseUrl: 'https://discuss.ropensci.org/',
+          discourseEmbedUrl: {{ .Page.Permalink | absURL }} };
```

Comment:
Maybe remove absURL?
In any case you can print the context .Page here (debugging link from an earlier comment) to see whaf variable to use?

Reply:
it was working before i'm pretty sure. that line of code `discourseEmbedUrl: {{ .Page.Permalink | absURL }} }` is the same, but maybe something else changes

Reply:
right! I see what happened, I'll fix it. The permalink variable isn't available from the partial so I'll need to pass it. Sorry!